images/win/scripts/ImageHelpers/InstallHelpers.ps1

```diff
@@ -1,124 +1,113 @@
-        $exitCode = $process.ExitCode
+    .EXAMPLE
+        Install-Binary -Url "https://go.microsoft.com/fwlink/p/?linkid=2083338" -Name "winsdksetup.exe" -ArgumentList ("/features", "+", "/quiet")
+    #>
 
-        if ($exitCode -eq 0 -or $exitCode -eq 3010)
-        {
-            Write-Host -Object 'Installation successful'
-            return $exitCode
-        }
-        else
-        {
-            Write-Host -Object "Non zero exit code returned by the installation process : $exitCode."
-            exit $exitCode
-        }
-    }
-    catch
-    {
-        Write-Host -Object "Failed to install the MSI $MsiName"
-        Write-Host -Object $_.Exception.Message
-        exit -1
-    }
-}
-
-
-function Install-EXE
-{
-    Param
-    (
-        [String]$Url,
-        [String]$Name,
-        [String[]]$ArgumentList
+    Param (
+        [Parameter(Mandatory)]
+        [String] $Url,
+        [Parameter(Mandatory)]
+        [String] $Name,
+        [String[]] $ArgumentList
     )
 
-    $exitCode = -1
+    Write-Host "Downloading $Name..."
+    $filePath = Start-DownloadWithRetry -Url $Url -Name $Name
 
-    try
+    # MSI binaries should be installed via msiexec.exe
+    $fileExtension = ([System.IO.Path]::GetExtension($Name)).Replace(".", "")
+    if ($fileExtension -eq "msi")
     {
-        Write-Host "Downloading $Name..."
-        $FilePath = "${env:Temp}\$Name"
-
-        Invoke-WebRequest -Uri $Url -OutFile $FilePath
+        $ArgumentList = ('/i', $filePath, '/QN', '/norestart')
+        $filePath = "msiexec.exe"
+    }
 
+    try
+    {
         Write-Host "Starting Install $Name..."
-        $process = Start-Process -FilePath $FilePath -ArgumentList $ArgumentList -Wait -PassThru
-        $exitCode = $process.ExitCode
+        $process = Start-Process -FilePath $filePath -ArgumentList $ArgumentList -Wait -PassThru
 
+        $exitCode = $process.ExitCode
         if ($exitCode -eq 0 -or $exitCode -eq 3010)
         {
-            Write-Host -Object 'Installation successful'
-            return $exitCode
+            Write-Host "Installation successful"
         }
         else
         {
-            Write-Host -Object "Non zero exit code returned by the installation process : $exitCode."
-            return $exitCode
+            Write-Host "Non zero exit code returned by the installation process: $exitCode"
         }
     }
     catch
     {
-        Write-Host -Object "Failed to install the Executable $Name"
+        Write-Host -Object "Failed to install the $fileExtension $Name"
         Write-Host -Object $_.Exception.Message
-        return -1
+        $exitCode = -1
     }
+
+    return $exitCode
 }
 
 function Stop-SvcWithErrHandling
-<#
-.DESCRIPTION
-Function for stopping the Windows Service with error handling
+{
+    <#
+    .DESCRIPTION
+        Function for stopping the Windows Service with error handling
 
-.AUTHOR
-Andrey Mishechkin v-andmis@microsoft.com
+    .AUTHOR
```

Comment:
Already 2 authors 😃or we can get rid of that section

Reply:
Yep, it's not supported by synopsis notation, so I removed it.